session.go

```diff
@@ -836,9 +836,23 @@ func (s *session) PrepareStmt(sql string) (stmtID uint32, paramCount int, fields
 		return
 	}
 
+	goCtx := goctx.Background()
+	inTxn := s.GetSessionVars().InTxn()
+	s.PrepareTxnCtx(goCtx)
 	prepareExec := executor.NewPrepareExec(s, executor.GetInfoSchema(s), sql)
 	err = prepareExec.DoPrepare()
-	return prepareExec.ID, prepareExec.ParamCount, prepareExec.Fields, errors.Trace(err)
+	if err != nil {
+		err = errors.Trace(err)
+		return
+	}
+	if !inTxn {
```

Comment:
Add comment:
// As we start a fake transaction to build the prepare executor before, we should rollback it here.